File: Majo/Act-Instructor_Henry/lenguajes/ruby/junio/26-06-24.rb
cont = 0
acum = 0
cont2 = 0
acum2 = 0
cant = 0
user = 0

while cant < 100000
    puts "Ingresa la cantidad que deseas aportar"
    var = gets.chomp.to_f
    cant = cant + var
    user = user + 1
    if var >= 5000
        cont = cont + 1
        acum = acum + var
    else
        cont2 = cont2 + 1
        acum2 = acum2 + var
    end

end 
puts "---------------------------------------------------------------------------"
puts " "
puts "-La cantidad total recogida es: $#{cant}."
puts " "
puts "-El promedio del total recogido es: $#{cant/user}"
puts " "
puts "---------------------------------------------------------------------------"
puts "-Personas que aportaron $5.000 o más: $#{cont}."
puts "-Cantidad total de los que aportaron $5.000 o más: $#{acum}."
puts "-Promedio de aporte de los que dieron $5.000 o más: $#{acum/cont}."
puts " "
puts "---------------------------------------------------------------------------" 
puts "-Personas que aportaron menos de $5.000: $#{cont2}."
puts "-Cantidad total de los que aportaron menos de $5.000: $#{acum2}."
puts "-Promedio de aporte de los que dieron menos de $5.000: $#{acum2/cont2}."
puts " "

File: Majo/Act-Instructor_Henry/lenguajes/dart/junio/26-06-24.dart
import 'dart:io';

void main(){

  int cont = 0;
  int acum = 0;
  int cont2 = 0;
  int acum2 = 0;
  int cant = 0;
  int user = 0;

  while (cant < 100000){
  
    print('ingrese la cantidad aportada por el aprendiz');
    int? a = int.parse(stdin.readLineSync()!);
    cant = cant + a;
    user = user + 1;

    if (a >= 5000) {
      cont = cont + 1;
      acum = acum + a;
    } else{
      cont2 = cont2 + 1;
      acum2 = acum2 + a;
    }
  }
  print('---------------------------------------------------------------------------');
  print(' ');
  print('-La cantidad total recogida es: \$$cant.');
  print(' ');
  print('-El promedio del total recogido es: \$${cant~/user}.');
  print(' ');
  print('---------------------------------------------------------------------------');
  print('-Personas que aportaron \$5.000  o más: $cont.');
  print('-Cantidad total de los que aportaron \$5.000 o más: \$$acum.');
  print('-Promedio de aporte de los que dieron \$5.000 o más: \$${acum~/cont}.');
  print(' ');
  print('---------------------------------------------------------------------------');
  print('-Personas que aportaron menos de \$5.000: $cont2.');
  print('-Cantidad de los que aportaron menos de \$5.000: \$$acum2.');
  print('-Promedio de aporte de los que dieron menos de \$5.000: \$${acum2~/cont2}.');
  print(' ');
  
}